### PR TITLE
py-beancount, fava: fixes and cleanup

### PR DESCRIPTION
**Pull request by mrdomino**
#### Description

Context in #4423. This addresses the issues mentioned by @reneeotten (namely: remove `noarch` from beancount, move setuptools to fava's `depends_lib`) and attempts to address @yan12125's point with one possible solution.

###### Type(s)

- [X] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS 10.14.5 18F132
Xcode 10.2.1 10E1001

###### Verification 
Have you

- [X] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [X] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [X] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [X] checked your Portfile with `port lint`?
- [ ] tried existing tests with `sudo port test`?
- [X] tried a full install with `sudo port -vst install`?
- [X] tested basic functionality of all binary files?

### CODE REVIEW

File: python/py-beancount/Portfile [hunk 1]
```diff
@@ -4,18 +4,22 @@ PortSystem          1.0
...
+revision            1
 checksums           rmd160  60650715322a8f302455f3028400281f23d850d2 \
                     sha256  ebcb59bd7c0e18a858c55d6c30eacee3fa949ee5d2c452a7aa80690e36ae2f77 \
                     size    597196
 
 license             GPL-2
 platforms           darwin
-supported_archs     noarch
-maintainers         nomaintainer
+maintainers         {blair @blair} openmaintainer
```
Comment (blair) [inline python/py-beancount/Portfile]: Please don't add me as a maintainer as I don't use this package any more.

File: python/py-beancount/Portfile [hunk 2]
```diff
@@ -4,18 +4,22 @@ PortSystem          1.0
 PortGroup           python 1.0
 
 name                py-beancount
+categories-append   finance
 version             2.2.1
+revision            1
```
Comment (blair) [inline python/py-beancount/Portfile]: Does the revision need to be bumped since the contents are not changing?

Comment (mf2k) [reply python/py-beancount/Portfile]: No, it does not.

**Comment by reneeotten** [comment]
I have no preferences with respect to which port is going to be removed/kept. If the decision is to keep ```python/py-beancount```, then you should make the ```finance/beancount``` port obsolete and set it to be ```replaced_by``` instead of directly removing it (see [the guide](https://guide.macports.org/#development.obsolete-portgroup)).

If you @mrdomino want to maintain the ports, please do add yourself as the maintainer. Adding ```openmaintainer``` as well signifies that you have no problem with others making "minor changes" to fix issues or updates.

File: python/py-beancount/Portfile (hunk 2 above)
Comment (mrdomino) [reply python/py-beancount/Portfile]: Ah, I'd thought that removing the supported_archs line might require a revision bump. If not, happy to remove it.

File: python/py-beancount/Portfile (hunk 1 above)
Comment (mrdomino) [reply python/py-beancount/Portfile]: Done.

File: python/py-beancount/Portfile (hunk 2 above)
Comment (mrdomino) [reply python/py-beancount/Portfile]: Removed it.

**Comment by mrdomino** [comment]
Okay, sorry about the noise; should be ready for another look.

**Comment by mrdomino** [comment]
@blair FYI the noarch removal is right above the maintainers line in py-beancount.

File: finance/beancount/Portfile [hunk 3]
```diff
@@ -1,35 +1,10 @@
...
-PortGroup           bitbucket 1.0
-
-bitbucket.setup     blais beancount bd3f97caca3d
+PortGroup           obsolete 1.0
 
 name                beancount
+replaced_by         py-beancount
 categories          finance
 license             GPL-2
 version             0.1
```
Comment (blair) [inline finance/beancount/Portfile]: Does this port need a revision bump now?

Comment (mrdomino) [reply finance/beancount/Portfile]: Seems so; bumped.

**Pull request merged**